### DISCUSSION TITLE
extend() and prefixUrl option

### DISCUSSION THREAD

**Discussion by montogeek** [sindresorhus/ky#419] [Q&A] [answered]
Hi!

Docs says that `extend` would inherit the default options of the extended instance. I have the following case:
```js
const client = ky.extend({
  prefixUrl: 'http://example.org',
  throwHttpErrors: false,
});
```

Then in other place I want to extend it:
```js
const apiClient = client.extend({
  prefixUrl: '/api'
})
```

I would expect that when calling `apiClient.get('users')`, it would make a request to `http://example.org/api/users`, but this it not the case, the request goes to `${currentHosname}/api/users`.

Is this not supported or I am missing something?

Thanks!

**Comment by Moshyfawn** [comment, marked as answer]
I don't think `.extend()` was meant to inherit the initial `prefixUrl` from `.create()` and prepend it to the `prefixUrl` specified. From the Ky readme on `.extend()`:
> Create a new ky instance with some defaults **overridden** with your own.

So, whatever options you specify in `.extend()` gonna **override** the ones from `.create()`.

You probably gonna need to pass an environment variable with your `BASE_URL` concatenated to the `prefixUrl` option to every `.extend()` to construct the URL you want.

**Reply by sholladay** [reply]
Yeah, exactly. think of `.extend()` as being like [object spread](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax), it's not doing anything terribly fancy. It just sets a new `prefixUrl`, which is not based on the previous `prefixUrl`.
